src/librustc/ty/maps.rs

```diff
@@ -180,6 +182,20 @@ impl<'tcx> Value<'tcx> for ty::SymbolName {
     }
 }
 
+struct QueryMap<D: QueryDescription> {
+    phantom: PhantomData<D>,
+    map: FxHashMap<D::Key, D::Value>,
```

Comment:
I like the idea but that would complicate the `define_map_struct` macro. Since all of this will be refactored in the near future anyway, I think it's OK to do this in a later PR.